src/algorithms/calorimetry/CalorimeterClusterRecoCoG.cc

```diff
@@ -417,8 +417,8 @@ edm4hep::MCParticle CalorimeterClusterRecoCoG::get_primary(const edm4hep::CaloHi
   //     can be improved!!
   edm4hep::MCParticle primary = contributor;
   while (primary.parents_size() > 0) {
+    if (primary.getGeneratorStatus() != 0) break;
     primary = primary.getParents(0);
-    if (primary.getGeneratorStatus() == 1) break;
```

Comment:
Oh, and the original ordering must have been the logical error. However even after this reordering I see status code 2, so I'd keep the new condition too.

![image](https://github.com/user-attachments/assets/bfa57780-b879-4dd0-90a8-07a42ae7f009)
(red: main, blue: this commit).